internal/providers/azure/azure.go

```diff
@@ -66,29 +63,43 @@ func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
 // FetchFromOvfDevice has the return signature of platform.NewFetcher. It is
 // wrapped by this and AzureStack packages.
 func FetchFromOvfDevice(f *resource.Fetcher, ovfFsTypes []string) (types.Config, report.Report, error) {
-	devicePath := filepath.Join(distro.DiskByIDDir(), configDeviceID)
-
+	var rawConfig []byte
 	logger := f.Logger
-	logger.Debug("waiting for config DVD...")
-	waitForCdrom(logger, devicePath)
 
-	fsType, err := checkOvfFsType(logger, devicePath, ovfFsTypes)
-	if err != nil {
-		return types.Config{}, report.Report{}, err
+	for _, ovfFsType := range ovfFsTypes {
+		device, err := execUtil.GetBlockDevices(ovfFsType)
+		if err != nil {
+			return types.Config{}, report.Report{}, fmt.Errorf("failed to retrieve block devices with FSTYPE=UDF: %v", err)
+		} else if len(device) > 0 {
+			for i, dev := range device {
+				rawConfig, err = getRawConfig(f, dev)
+				if len(rawConfig) > 0 && isCdromPresent(logger, dev) {
+					break
+				} else if err != nil && i == len(device) {
+					return types.Config{}, report.Report{}, fmt.Errorf("failed to retrieve config: %v", err)
+				}
+			}
+		}
+
 	}
+	return util.ParseConfig(logger, rawConfig)
```

Comment:
Suggestion: move this call inside the loop, to the point where we get the config.  That'll let you simplify the logic in a couple places.